### ISSUE TITLE
Docker Installation -- What now?

### ISSUE THREAD

**Issue by steele232** [remacs/remacs#1386]
Hello, 

I'm new to this project and I am trying to run Remacs for the first time. I opted for the Docker installation and I got that to build and run, but now I don't know how to "get into it." (that's in quotes because I recognize that's a little vague; I'm hoping to see a new window pop up running emacs and then I want to edit text documents in it)

When I do `docker ps` I can see that the image is running. This the result: 
```
CONTAINER ID        IMAGE                 COMMAND             CREATED             STATUS              PORTS               NAMES
65fa8ba7c817        remacs-build:latest   "/bin/bash"         13 minutes ago      Up 13 minutes                           remacs_remacs_1
```

I see that after the installation section of the README, there is a block that talks about running it and it says:
```
# Using -q to ignore your .emacs.d, so Remacs starts up quickly.
# RUST_BACKTRACE is optional, but useful if your instance crashes.
$ RUST_BACKTRACE=1 src/remacs -q
```

It tried running that last command and it says 
```
-bash: src/remacs: No such file or directory
```

I'm in the same directory as where I cloned the repo and ran the dockerfile with docker-compose, so I assumed that would work. As I examined the filesystem, I confirmed that there is no `remacs` executable to be found. I looked in a few other folders and didn't find it, so now I'm here asking for help. 

How can I start using Remacs?

**Comment by shaleh** [comment]
Great question. To date we have used Docker a way to built and run tests. Do you want the GUI or just a text mode interface? What OS are you trying to run this on? Because a Mac most likely won't run the Docker's Linux GUI without also running a X server on the Mac. Either way the Docker would need to be able to contact the host's X server.

For the text mode UI you should be able to use docker to run a shell and then fire up Remacs. But then it will only be able to edit files exposed to the Docker instance.

**Comment by steele232** [comment]
I was wanting to use the GUI (I think). I'm running it on MacOS. I currently don't know anything about running an X server.

I think I could be okay with only being able to edit files exposed to the Docker instance (maybe?) 

OK. Taking a step back. I'm not quite sure what the difference is between the text mode UI and the GUI is. I've used emacs from a terminal; is that what the text mode UI would be like? If so, I could be okay with that. If that is the case, could you help me "use docker to run a shell and then fire up Remacs"? My docker skills are kind of mediocre. 

My idea is to do [something like this](https://askubuntu.com/questions/505506/how-to-get-bash-or-ssh-into-a-running-container-in-background-mode): 

````
sudo docker attach c03a94af3e8d
```

I just ran that command and I believe I'm shelled into the container now, but now I'm not sure what to do. The file directory looks just like my old directory and so I don't see where to run Remacs

```
root@c03a94af3e8d:/Users/jonathansteele/rusty/remacs# remacs
bash: remacs: command not found
```

Any new ideas, or different directions I should go?

**Comment by shaleh** [comment]
Docker on MacOS is based on a Linux virtual machine. The container is a view into that Linux OS. This is why you cannot easily make a window appear. The Linux idea of a window system and the Mac idea are not the same. If you have run emacs from inside a terminal window before and were happy with that interface then what you will get from Docker easily will also make you happy.

The `remacs` binary is in the `src` directory if you have a successful build. Trying running `make` from that attached shell in the root of the remacs source tree and see what happens.

My personal opinion is you would have a better experience running remacs natively on your Mac. However, this is a good opportunity to learn and experiment with Docker.

**Comment by steele232** [comment]
Ahh... Thank you. I ran `make` in the container and it performed the build (for more than half an hour) and then I found the `remacs` binary in the `src` directory just as you said. I was able to run it and edit and play around a bit. That was really good, thank you. 

I think you're right that I would have a better experience if I just built it and ran it natively, that would also let me run spacemacs on top of it as well. If I run into problems, I could raise another issue. Is there a different way that I **should** be communicating that would make it easier for you or **is this the proper channel?**
